cmd/protoc-gen-docs/htmlGenerator.go

```diff
@@ -1111,6 +1113,27 @@ func getFieldBehavior(options *descriptor.FieldOptions) []annotations.FieldBehav
 	return s
 }
 
+func getRequiredFromComments(loc protomodel.LocationDescriptor) bool {
+	com := loc.GetLeadingComments()
+	if com == "" {
+		com = loc.GetTrailingComments()
+		if com == "" {
+			return false
+		}
+	}
+
+	if strings.Contains(com, "Required") {
+		return true
+	}
+
+	if strings.Contains(com, "Optional") {
+		return true
+	}
+
+	// default is required.
+	return true
```

Comment:
Comment above.

Reply:
So, we do similar to generateComment() in that we first use the LeadingComments and if empty, use Trailing comments. We don't actually try and use both.

If both are empty, we return false which makes it such that no comments means not required. 
If it contains Optional, then we also say not required. 
If the text contains Required/REQUIRED, we will say yes, however, we also say yes if the text does not contain Required/REQUIRED. This implies that there is no need for the contains as any text means required:yes.
I'm thinking you really want this return to be false, else you can remove the contains check.